src/main/java/com/spring/studentmanagement/services/CourseService.java

```diff
@@ -0,0 +1,21 @@
+package com.spring.studentmanagement.services;
+
+import com.spring.studentmanagement.exceptions.CourseNotFoundException;
+import com.spring.studentmanagement.exceptions.UserNotFoundException;
+import com.spring.studentmanagement.models.AppUser;
+import com.spring.studentmanagement.models.Course;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Transactional
```

Comment:
Transactional annotation must be on the implementation, not on the service.
When we use transactional at the class level, everything in that class will happen inside a transaction.
We can also use transactional at the method level.